### ISSUE TITLE
Socket.InternalShutdown is ignoring LingerOption

### ISSUE THREAD

**Issue by kaiohenrique** [dotnet/runtime#104000]
### Description

We have set LingerOption to avoid TIME_WAIT socket state but it seems not change the behavior. Extending Network stream and only calling socket Close and Dispose seems to be doing the trick. No sure why Socket.InternalShutdown is impedind the LingerOption to work as expected. Am I missing anything here?
Note: in the the request headers, we are sending Connect: close to avoid pooling

### Reproduction Steps

SocketsHttpHandler ConnectCallback  overwrite
```
 ConnectCallback = async (context, cts) =>
 {
     var socket = new Socket(SocketType.Stream, ProtocolType.Tcp)
     {
         NoDelay = apiConfig.Socket.NoDelay,
         LingerState = new LingerOption(true, 0)
     };

     try
     {                                
         await socket.ConnectAsync(context.DnsEndPoint, cts);
         return new SocketStream(socket, ownsSocket: false);
     }
     catch (Exception e)
     {
         socket.Dispose();
         throw;
     }
   
 }
```

```
 public class SocketStream : NetworkStream
 {        
     private Socket _socket;        
     public SocketStream(Socket socket, bool ownsSocket) : base(socket, ownsSocket)
     {            
         _socket = socket;            
     }
     private  object call(object o, string methodName, params object[] args)
     {
         var mi = o.GetType().GetMethod(methodName, System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance);
         if (mi != null)
         {
             return mi.Invoke(o, args);
         }
         return null;
     }

     protected override void Dispose(bool disposing)
     {
         //call(_socket, "InternalShutdown", SocketShutdown.Both); to reproduce the TIME_WAIT you should not comment this line

         _socket.Close();
         _socket.Dispose();
         base.Dispose(disposing);                    
     }
 }
```

### Expected behavior

NetworkStream should respect LingerOptions

### Actual behavior

As NetworkStream is calling Socket.InternalShutdown, LingerOption is not being respected and socket is leaking in TIME_WAIT state

### Regression?

Only tested in .Net8 both in Windows and Linux

**Comment by wfurt** [comment]
Why do you set `ownsSocket: false`? AFAIK the behavior can change based on the ownership. cc @tmds

**Comment by kaiohenrique** [comment]
Hi, @wfurt 
In this case, I want to avoid the NetworkStream.Dispose to execute the [Socket.InternalShutdown method](https://github.com/dotnet/runtime/blob/release/8.0/src/libraries/System.Net.Sockets/src/System/Net/Sockets/NetworkStream.cs#L391). Overriding the NetworkStream.Dispose method and only performing the Socket.Close and Socket.Dispose seems to workaround the issue and LingerOption is correctly applied

**Comment by kaiohenrique** [comment]
Hi, @wfurt 
I need help to understand if avoiding [Socket.InternalShutdown](https://github.com/dotnet/runtime/blob/release/8.0/src/libraries/System.Net.Sockets/src/System/Net/Sockets/NetworkStream.cs#L391) is a valid workaround so LingerOption works as expected. In our tests, avoiding the call to Socket.InternalShutdown got rid of connections in time_wait state completly. Are we missing any side effect?

**Comment by kaiohenrique** [comment]
Update:
I have seen this behavior
_socket.Shutdown(SocketShutdown.Receive); //avoid time_wait and LingerOption is respected 
_socket.Shutdown(SocketShutdown.Send); //lots of sockets in time_wait
_socket.Shutdown(SocketShutdown.Both); //lots of sockets in time_wait
_socket.Close();
_socket.Dispose();

**Comment by liveans** [comment]
Triage: Not critical for 9.0. Moving to the future.

**Comment by tmds** [comment]
> Are you able to reproduce adding a Task.Delay after the loop?

No, also when the program doesn't exit, setting the `new LingerOption(true, 0)` option results in there being no `TIME_WAIT` sockets.

I can't reproduce the issue.
